var/spack/repos/builtin/packages/hermes/package.py

```diff
@@ -11,38 +11,129 @@ class Hermes(CMakePackage):
+    )
+    version(
+        "0.9.9-beta",
+        url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.9-beta.tar.gz",
+        sha256="d2e0025a9bd7a3f05d3ab608c727ed15d86ed30cf582549fe996875daf6cb649",
+    )
+    version(
+        "0.9.8-beta",
+        url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.8-beta.tar.gz",
+        sha256="68e9a977c25c53dcab7d7f6ef0df96b2ba4a09a06aa7c4a490c67faa2a78f077",
+    )
+    version(
+        "0.9.5-beta",
+        url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.5-beta.tar.gz",
+        sha256="f48d15591a6596e8e54897362ec2591bc71e5de92933651f4768145e256336ca",
+    )
     version(
         "0.9.0-beta",
         url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.0-beta.tar.gz",
         sha256="abf258a52fa79729dfeb28559957abf8945f3ad37cadefb3bc685227c5f057a8",
     )
 
+    variant("adios", default=False, description="Build Adios tests")
+    variant("ares", default=False, description="Enable full libfabric install")
+    variant("compress", default=False, description="Enable compression")
+    variant("encrypt", default=False, description="Enable encryption")
+    variant("mpiio", default=True, description="Enable MPI I/O adapter")
+    variant("python", default=False, description="Build Python Wrapper")
+    variant("stdio", default=True, description="Enable STDIO adapter")
+    variant("vfd", default=False, description="Enable HDF5 VFD")
+    variant("zmq", default=False, description="Build ZeroMQ tests")
+
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
+    depends_on("libelf")
 
-    variant("vfd", default=False, description="Enable HDF5 VFD")
+    depends_on("hermes-shm@master+boost+cereal+mochi")
 
-    depends_on("mochi-thallium~cereal@0.8:")
-    depends_on("catch2")
-    depends_on("glpk")
-    depends_on("glog@0.4.0:")
-    depends_on("mpi")
-    depends_on("hdf5@1.14.0:", when="+vfd")
-    depends_on("yaml-cpp")
+    depends_on("hermes-shm+adios", when="+adios")
+    depends_on("hermes-shm+ares", when="+ares")
+    depends_on("hermes-shm+compress", when="+compress")
+    depends_on("hermes-shm+encrypt", when="+encrypt")
+    depends_on("hermes-shm+mpiio", when="+mpiio")
+    depends_on("hermes-shm+vfd", when="+vfd")
+    depends_on("hermes-shm+zmq", when="+zmq")
+
+    depends_on("py-jarvis-util", type="test")
 
     def cmake_args(self):
-        args = [
-            self.define("HERMES_RPC_THALLIUM", True),
-            self.define("HERMES_INSTALL_TESTS", True),
-            self.define("BUILD_TESTING", True),
-            self.define_from_variant("HERMES_ENABLE_VFD", "vfd"),
-        ]
+        args = []
+        if "+mpiio" in self.spec:
+            args.append("-DHERMES_ENABLE_MPIIO_ADAPTER=ON")
+            if "openmpi" in self.spec:
+                args.append("-DHERMES_OPENMPI=ON")
+            elif "mpich" in self.spec:
+                args.append("-DHERMES_MPICH=ON")
```

Comment:
Trying to build this, I saw that does not work (with hermes@1.2.1):

When -DHERMES_ENABLE_MPIIO_ADAPTER=ON is defined, and the none of the other two is defined, the build files with:
```py
spack-src/hermes_adapters/mpiio/mpiio_fs_api.h:696:2: error: #error "No MPI implementation specified for MPIIO adapter"
```
This is the spec `hermes@1.2.1%gcc@13.1.0~adios~ares~compress~encrypt~ipo+mpiio~python+stdio~vfd~zmq`

It only has `+mpiio`, but does not have `openmpi` or `mpich` because those are not variants of `hermes` and you cannot add them to the spec without that, AFAIK.

AFAICS, You need to select which MPI to compile with, and of course `hermes-shm+mpiio` should use the same MPI. Apparently, this seems to be a way to get the MPI that is selected using the "mpi" alias:
```py
var/spack/repos/builtin/packages/r-rmpi/package.py:        mpi_name = spec["mpi"].name
var/spack/repos/builtin/packages/r-rmpi/package.py:        if mpi_name == "openmpi":
var/spack/repos/builtin/packages/r-rmpi/package.py:        elif mpi_name == "mpich":
g
```

Reply:
I found that `spec["..."].name is also used for getting the selected `blas` in other recipes.

Pushed an initial attempt.

But as it needs a further fix, I'll mark this as draft until ready to prevent GitLab CI from spending paid compute Pods on building it.